File: MATLAB/initialize_flow_variables.m
function[density,temperature,velocity] = initialize_flow_variables(spacesteps,timesteps,delta_x)
    %Create Placeholder 
    density = zeros(timesteps,spacesteps);
    temperature = zeros(timesteps,spacesteps);
    velocity = zeros(timesteps,spacesteps);
    
    %Create intial values which help with better convergence
    
    for i = 1:spacesteps
        x = (i-1)*delta_x;
        density(1,i) = 1-(0.3146*x); 
        temperature(1,i) = 1-(0.2314*x);
        velocity(1,i) = (0.1+(1.09*x)).*(temperature(1,i)).^0.5;
        
    end
    
end

File: PYTHON/initialize_flow_variables.py
import numpy as np
from boundary_condition import endpoints

def initialize_with_zeros(timesteps,spacesteps):
    density = np.zeros((timesteps,spacesteps))
    temperature = np.zeros((timesteps,spacesteps))
    velocity = np.zeros((timesteps,spacesteps))
    # We create a placeholder for our data
    return density,temperature,velocity

def initialize_with_values(delta_x,timesteps,spacesteps):
    (density , temperature , velocity)= initialize_with_zeros(int(timesteps),int(spacesteps)) 
    #We provide linearly varying values to out flow variables at time = 0 to ensure easy convergence
    for i in range(spacesteps):
        density[0,i] = 1 - 0.3146*delta_x*i
        temperature[0,i] = 1 - 0.2314*delta_x*i
        velocity[0,i] = (0.1 + 1.09*delta_x*i)*temperature[0,i]**(0.5)
        
    density,temperature,velocity = endpoints(density,temperature,velocity)
    return density,temperature,velocity                                         
